src/exporter.py

```diff
@@ -141,6 +142,27 @@ def scrape(self):
             self.track_timed_out_workers()
         self.track_queue_metrics()
 
+    @lru_cache(maxsize=32)
+    def find_queue_by_task(self, target: Task) -> str:
+        """Provider a queue name based on metadata coming from eiether a worker or a task being processed by it"""
+
+        try:
+            queue_name = None
+            # https://github.com/celery/celery/issues/5321
+            # task_info = self.app.control.inspect().query_task(task.id)
+            # As the received tasks are considered as active ones
+            task_set = self.app.control.inspect().registered().get(target.hostname)
+            task_info = [task for task in task_set if task['id'] == target.id].pop()
+            queue_name = task_info["delivery_info"]["routing_key"]
+        except TimeoutError as error:
+            # the broker doesn't respond
+            logger.error(f"Couldn't fetch the task info of {target.id}: {error.strerror}")
+        except IndexError as error:
+            # couldn't find the target task by id
+            # the dictionary path is missing
+            logger.warning(f"Couldn't find the target task by its id: {target.id}")
+        return queue_name
```

Comment:
What happens when an app deployment changes the queue and the celery-exporter doesn't restart?

Reply:
It will keep return the old queue name until the cache will have been invalidated. 
That was my next question which I missed to include in the header. Pardon for that :pray:

How to not query the broker for each task as it can lead to some performance degradation depending on the producing rate ?   

Several strategies comes at once, namely:
* we need to start another worker to invalidate the cache 
* remove caching for this lookup method at this point 
* decrease the cache size

I've also taken a look into Flower to see how it manages such cases, but couldn't find any relevant parts related to queue name processing.